nifi-nar-bundles/nifi-rethinkdb-bundle/nifi-rethinkdb-processors/src/main/java/org/apache/nifi/processors/rethinkdb/GetRethinkDB.java

```diff
@@ -0,0 +1,197 @@
+            .required(true)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(ResultType.STRING, true))
+            .expressionLanguageSupported(true)
+            .build();
+
+    protected String READ_MODE_KEY = "read_mode";
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    public static final String RETHINKDB_ERROR_MESSAGE = "rethinkdb.error.message";
+
+    protected Gson gson = new Gson();
+
+    static {
+        final Set<Relationship> tempRelationships = new HashSet<>();
+        tempRelationships.add(REL_SUCCESS);
+        tempRelationships.add(REL_FAILURE);
+        tempRelationships.add(REL_NOT_FOUND);
+        relationships = Collections.unmodifiableSet(tempRelationships);
+
+        final List<PropertyDescriptor> tempDescriptors = new ArrayList<>();
+        tempDescriptors.add(DB_NAME);
+        tempDescriptors.add(DB_HOST);
+        tempDescriptors.add(DB_PORT);
+        tempDescriptors.add(USERNAME);
+        tempDescriptors.add(PASSWORD);
+        tempDescriptors.add(TABLE_NAME);
+        tempDescriptors.add(CHARSET);
+        tempDescriptors.add(RETHINKDB_DOCUMENT_ID);
+        tempDescriptors.add(READ_MODE);
+        tempDescriptors.add(MAX_DOCUMENTS_SIZE);
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        Charset charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(flowFile).getValue());
+        String id = context.getProperty(RETHINKDB_DOCUMENT_ID).evaluateAttributeExpressions(flowFile).getValue();
+        String readMode = context.getProperty(READ_MODE).evaluateAttributeExpressions(flowFile).getValue();
+
+        if ( StringUtils.isEmpty(id) ) {
+            getLogger().error("Empty id '" + id + "'");
+            flowFile = session.putAttribute(flowFile, RETHINKDB_ERROR_MESSAGE, "Empty id '" + id + "'");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        try {
+            long startTimeMillis = System.currentTimeMillis();
+            Map<String,Object> document = getDocument(id, readMode);
+
+            if ( document == null ) {
+                getLogger().error("Document not found for id " + id);
```

Comment:
Is not found an error?  It seems possible that this could happen at high volume.  Given that it has it's own relationship, and there is a separate relationship for failure, didn't this "succeed" in not finding the key?  I don't have a strong opinion, I'll accept it as-is if you believe in it.